src/Security/AccessRightsPolicy.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Security;
+
+use App\Entity\Activity;
+use App\Entity\User;
+use App\Repository\ActivityUserRepository;
+
+class AccessRightsPolicy
+{
+
+    /**
+     * @var ActivityUserRepository
+     */
+    private $repository;
+
+    public function __construct(ActivityUserRepository $repository)
```

Comment:
Use full variable name `$activityUserRepository` so later you know for which entity is this repository without scrolling to the constructor.